modules/system/utility-wfrp4e.js

```diff
@@ -1335,9 +1336,49 @@ export default class WFRP_Utility {
     return path.replace("tokens/popout/", "tokens/");
   }
 
-  static sleep(ms)
-  {
-    return new Promise(resolve => setTimeout(resolve, ms));
+  static async sleep(ms) {
+    await new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  static getActorOwner(actor) { 
+    if (actor.hasPlayerOwner) {
+      for (let u of game.users.contents.filter(u => u.active && !u.isGM)) {
+        if (u.character?.id === actor.id) {
+          return u;
+        }
+      }
+      for (let u of game.users.contents.filter(u => u.active && !u.isGM)) {
+        if (actor.ownership.default >= CONST.DOCUMENT_OWNERSHIP_LEVELS.OWNER || actor.ownership[u.id] >= CONST.DOCUMENT_OWNERSHIP_LEVELS.OWNER) {
+        return u;
+        }
+      }
+    }
+    return game.users.contents.find(u => u.active && u.isGM);
+  }
+
+  static async awaitSocket(owner, type, payload, content) {
+    let msg = await WFRP_Utility.createSocketRequestMessage(owner, content);
+    payload.socketMessageId = msg.id;
+    game.socket.emit("system.wfrp4e", {
+      type: type,
+      payload: payload
+    });
+    do {
+      await WFRP_Utility.sleep(250);
+      msg = game.messages.get(msg.id);
+    } while (msg);
+  }
+
+  static async createSocketRequestMessage(owner, content) {
+    let chatData = {
+      content: `<p class='requestmessage'><b><u>${owner.name}</u></b>: ${content}</p?`,
+      whisper: ChatMessage.getWhisperRecipients("GM")
```

Comment:
So, for that there is a css rule to hide it. i think i tried with ```renderChatMessage``` but then - it wasn't added to messages collection at all. So the messages will not be shown however they will be included in DOM.

Reply:
You shouldn't need a CSS rule, simply 

`ui.chat.element.find("[data-message-id='TR3EtP2km4RlcDAl']").remove()`

For example, and this is not a database operation, `game.messages` still contains everything.

Reply:
If that's a better way to do it, I will try and add the debug thing

Reply:
that's a way to do it from console, I'm not sure if it'll work in `renderChatMessage`, so it may need some experimenting. 

I see your CSS rule, I was unaware of that syntax (I didn't think you could hide the whole message based on a child class). Though, I still think it would be useful to have in dev mode.

Reply:
i have added a function as you described however it doesn't seem to help, therefore i've left the css rule. the problem is that i cannot remove the whole html from message.